Change in unit 3 for model selection

NYUVideoLab opened the pull request:
Hi, Sundeep,
I modified the slide quite a bit. I changed somewhat how bias and variance tradeoffs are presented. Removed the sentence / slide that says that we will not consider the case with noise.  I also added a few last slides about using the same approach for feature selection (in stead of polynomial order selection) and introduced several different ways for feature section (exhaustive search, forward stepwise, and backward stepsize, and using correlation). I thought that this may be necessary as the lab actually does feature selection. I was afraid that the students may not know what to do if i do not make such connections.

For the demo, i found that using 40 samples and noise variance of 0.2 can give quite different results, each time i run the demo program. So i changed to using 100 samples and noise variance of 0.4 (so that we can see the variance more). For the standard error, I do not quite understand why you divide the SDV by sqrt (K-1). I think the correct way should be *sqrt (K/(K-1)). But instead, I just used STD. If i scale done by sqrt(K-1), the SE becomes so small with 100 samples. 

For the lab, I decided to split your lab03 to two parts. The first part (lab03) will only do correlation based feature section. I deleted the part on LASSO.  I will use your material to crease lab03b, which will do the LASSO part for the next week.

NYUVideoLab committed:
Make minor correction to the One-hot coding table on slide 42

NYUVideoLab committed:
replace q2 by q3

NYUVideoLab committed:
Modified the way the bias-variance tradeoff is presented somewhat. Also added a few slides about feature selection. This is necessary since Lab is actually for feature selection.

NYUVideoLab committed:
Changed the number of samples to 100 so that the results are not too much influenced by the random noise added so that each time we run the program we get similar results. Also changed the Standard Error Calculation to just use the std.

NYUVideoLab committed:
Removed the part for using LASSO. Will add another lab called lab03b which uses LASSO for feature selection.

sdrangan commented:
Can I talk to you about this in a couple of hours.  Maybe easier to
explain.  Just in airport now

On Mon, Feb 5, 2018, 11:25 Yao Wang <notifications@github.com> wrote:

> Hi, Sundeep,
> I modified the slide quite a bit. I changed somewhat how bias and variance
> tradeoffs are presented. Removed the sentence / slide that says that we
> will not consider the case with noise. I also added a few last slides about
> using the same approach for feature selection (in stead of polynomial order
> selection) and introduced several different ways for feature section
> (exhaustive search, forward stepwise, and backward stepsize, and using
> correlation). I thought that this may be necessary as the lab actually does
> feature selection. I was afraid that the students may not know what to do
> if i do not make such connections.
>
> For the demo, i found that using 40 samples and noise variance of 0.2 can
> give quite different results, each time i run the demo program. So i
> changed to using 100 samples and noise variance of 0.4 (so that we can see
> the variance more). For the standard error, I do not quite understand why
> you divide the SDV by sqrt (K-1). I think the correct way should be *sqrt
> (K/(K-1)). But instead, I just used STD. If i scale done by sqrt(K-1), the
> SE becomes so small with 100 samples.
>
> For the lab, I decided to split your lab03 to two parts. The first part
> (lab03) will only do correlation based feature section. I deleted the part
> on LASSO. I will use your material to crease lab03b, which will do the
> LASSO part for the next week.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/sdrangan/introml/pull/21
> Commit Summary
>
>    - Make minor correction to the One-hot coding table on slide 42
>    - replace q2 by q3
>    - Modified the way the bias-variance tradeoff is presented somewhat.
>    Also added a few slides about feature selection. This is necessary since
>    Lab is actually for feature selection.
>    - Changed the number of samples to 100 so that the results are not too
>    much influenced by the random noise added so that each time we run the
>    program we get similar results. Also changed the Standard Error Calculation
>    to just use the std.
>    - Removed the part for using LASSO. Will add another lab called lab03b
>    which uses LASSO for feature selection.
>
> File Changes
>
>    - *M* lectures/Lect03_MultLinRegression.pdf
>    <https://github.com/sdrangan/introml/pull/21/files#diff-0> (0)
>    - *M* lectures/Lect03_MultLinRegression.pptx
>    <https://github.com/sdrangan/introml/pull/21/files#diff-1> (0)
>    - *M* lectures/Lect04_ModelSelection.pdf
>    <https://github.com/sdrangan/introml/pull/21/files#diff-2> (0)
>    - *M* lectures/Lect04_ModelSelection.pptx
>    <https://github.com/sdrangan/introml/pull/21/files#diff-3> (0)
>    - *M* unit02_mult_lin_reg/lab02_robot_calib_partial.ipynb
>    <https://github.com/sdrangan/introml/pull/21/files#diff-4> (32)
>    - *M* unit03_model_sel/demo03_1_polyfit.ipynb
>    <https://github.com/sdrangan/introml/pull/21/files#diff-5> (221)
>    - *M* unit03_model_sel/lab03_neural_partial.ipynb
>    <https://github.com/sdrangan/introml/pull/21/files#diff-6> (137)
>    - *A* unit03_model_sel/lab03_neural_soln.ipynb
>    <https://github.com/sdrangan/introml/pull/21/files#diff-7> (664)
>
> Patch Links:
>
>    - https://github.com/sdrangan/introml/pull/21.patch
>    - https://github.com/sdrangan/introml/pull/21.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sdrangan/introml/pull/21>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ANO8I6vwIFhuaG3nyScAzRi6XG2o9h2Bks5tRnS7gaJpZM4R431C>
> .
>

NYUVideoLab committed:
Revising Standard Error definition

NYUVideoLab committed:
Changing the SE calculation to add back the division by sqrt(nfold-1)

NYUVideoLab commented:
May be tomorrow morning? (My cell is 732-939-3399, any time after 8AM should be fine)

I just found the answer about STANDARD ERROR. So I reverted back to what you had.
___________________________________________________
Yao Wang
Professor of Electrical and Computer Engineering
Tandon School of Engineering
New York University
yaowang@nyu.edu
http://vision.poly.edu



> On Feb 4, 2018, at 10:33 PM, sdrangan <notifications@github.com> wrote:
> 
> Can I talk to you about this in a couple of hours. Maybe easier to
> explain. Just in airport now
> 
> On Mon, Feb 5, 2018, 11:25 Yao Wang <notifications@github.com> wrote:
> 
> > Hi, Sundeep,
> > I modified the slide quite a bit. I changed somewhat how bias and variance
> > tradeoffs are presented. Removed the sentence / slide that says that we
> > will not consider the case with noise. I also added a few last slides about
> > using the same approach for feature selection (in stead of polynomial order
> > selection) and introduced several different ways for feature section
> > (exhaustive search, forward stepwise, and backward stepsize, and using
> > correlation). I thought that this may be necessary as the lab actually does
> > feature selection. I was afraid that the students may not know what to do
> > if i do not make such connections.
> >
> > For the demo, i found that using 40 samples and noise variance of 0.2 can
> > give quite different results, each time i run the demo program. So i
> > changed to using 100 samples and noise variance of 0.4 (so that we can see
> > the variance more). For the standard error, I do not quite understand why
> > you divide the SDV by sqrt (K-1). I think the correct way should be *sqrt
> > (K/(K-1)). But instead, I just used STD. If i scale done by sqrt(K-1), the
> > SE becomes so small with 100 samples.
> >
> > For the lab, I decided to split your lab03 to two parts. The first part
> > (lab03) will only do correlation based feature section. I deleted the part
> > on LASSO. I will use your material to crease lab03b, which will do the
> > LASSO part for the next week.
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/sdrangan/introml/pull/21
> > Commit Summary
> >
> > - Make minor correction to the One-hot coding table on slide 42
> > - replace q2 by q3
> > - Modified the way the bias-variance tradeoff is presented somewhat.
> > Also added a few slides about feature selection. This is necessary since
> > Lab is actually for feature selection.
> > - Changed the number of samples to 100 so that the results are not too
> > much influenced by the random noise added so that each time we run the
> > program we get similar results. Also changed the Standard Error Calculation
> > to just use the std.
> > - Removed the part for using LASSO. Will add another lab called lab03b
> > which uses LASSO for feature selection.
> >
> > File Changes
> >
> > - *M* lectures/Lect03_MultLinRegression.pdf
> > <https://github.com/sdrangan/introml/pull/21/files#diff-0> (0)
> > - *M* lectures/Lect03_MultLinRegression.pptx
> > <https://github.com/sdrangan/introml/pull/21/files#diff-1> (0)
> > - *M* lectures/Lect04_ModelSelection.pdf
> > <https://github.com/sdrangan/introml/pull/21/files#diff-2> (0)
> > - *M* lectures/Lect04_ModelSelection.pptx
> > <https://github.com/sdrangan/introml/pull/21/files#diff-3> (0)
> > - *M* unit02_mult_lin_reg/lab02_robot_calib_partial.ipynb
> > <https://github.com/sdrangan/introml/pull/21/files#diff-4> (32)
> > - *M* unit03_model_sel/demo03_1_polyfit.ipynb
> > <https://github.com/sdrangan/introml/pull/21/files#diff-5> (221)
> > - *M* unit03_model_sel/lab03_neural_partial.ipynb
> > <https://github.com/sdrangan/introml/pull/21/files#diff-6> (137)
> > - *A* unit03_model_sel/lab03_neural_soln.ipynb
> > <https://github.com/sdrangan/introml/pull/21/files#diff-7> (664)
> >
> > Patch Links:
> >
> > - https://github.com/sdrangan/introml/pull/21.patch
> > - https://github.com/sdrangan/introml/pull/21.diff
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/sdrangan/introml/pull/21>, or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/ANO8I6vwIFhuaG3nyScAzRi6XG2o9h2Bks5tRnS7gaJpZM4R431C>
> > .
> >
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/sdrangan/introml/pull/21#issuecomment-362972899>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AhRjPHd16AC-cxGo32nGjHinlPC43su_ks5tRnZ1gaJpZM4R431C>.
>

NYUVideoLab committed:
updated labs

NYUVideoLab committed:
Change lab03 to lab03a

NYUVideoLab committed:
Update sequence.md

Change Lab3 to Lab3a. Will add Lab3b later for LASSO based feature selection.

NYUVideoLab committed:
minor language change

NYUVideoLab committed:
Merge branch 'master' of https://github.com/yaowangatpoly/introml

NYUVideoLab committed:
correcting typo

NYUVideoLab committed:
Added description about Ridge Regression. Added material regarding other feature selection methods.

NYUVideoLab committed:
Added feature ranking based on correlation, f-test, and mutual information, and RFE, and comparison with LASSO

NYUVideoLab committed:
Added a homework on LASSO and feature selection

NYUVideoLab committed:
Update sequence.md

NYUVideoLab committed:
Added some comments about cross validation at the end

NYUVideoLab committed:
added a comment that the cross-validation results depend on the random shuffling when the data amount is small

NYUVideoLab committed:
Merge branch 'master' of https://github.com/yaowangatpoly/introml

NYUVideoLab commented:
Sundeep, I made additional changes to part b of unit 3, as i mentioned in the email.

NYUVideoLab committed:
New lab for feature selection using student-performance data set

NYUVideoLab committed:
Update sequence.md

NYUVideoLab committed:
Update sequence.md

NYUVideoLab committed:
Update sequence.md
